pkg/controllers/disruption/consolidation_test.go

```diff
@@ -749,507 +867,694 @@ var _ = Describe("Consolidation", func() {
+								Kind:               "ReplicaSet",
+								Name:               rs.Name,
+								UID:                rs.UID,
+								Controller:         ptr.Bool(true),
+								BlockOwnerDeletion: ptr.Bool(true),
+							},
 						},
 					},
-				},
-				ResourceRequirements: v1.ResourceRequirements{
-					Requests: v1.ResourceList{
-						v1.ResourceCPU: resource.MustParse("2"),
+					ResourceRequirements: v1.ResourceRequirements{
+						Requests: v1.ResourceList{
+							v1.ResourceCPU: resource.MustParse("2"),
+						},
 					},
-				},
-			})
-			nodeClaim2, node2 := test.NodeClaimAndNode(v1beta1.NodeClaim{
-				ObjectMeta: metav1.ObjectMeta{
-					Labels: map[string]string{
-						v1beta1.NodePoolLabelKey:     nodePool.Name,
-						v1.LabelInstanceTypeStable:   mostExpensiveInstance.Name,
-						v1beta1.CapacityTypeLabelKey: mostExpensiveOffering.CapacityType,
-						v1.LabelTopologyZone:         mostExpensiveOffering.Zone,
+				})
+				nodeClaim2, node2 := test.NodeClaimAndNode(v1beta1.NodeClaim{
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: map[string]string{
+							v1beta1.NodePoolLabelKey:     nodePool.Name,
+							v1.LabelInstanceTypeStable:   mostExpensiveInstance.Name,
+							v1beta1.CapacityTypeLabelKey: mostExpensiveOffering.CapacityType,
+							v1.LabelTopologyZone:         mostExpensiveOffering.Zone,
+						},
 					},
-				},
-				Status: v1beta1.NodeClaimStatus{
-					ProviderID: test.RandomProviderID(),
-					Allocatable: map[v1.ResourceName]resource.Quantity{
-						v1.ResourceCPU:  resource.MustParse("5"),
-						v1.ResourcePods: resource.MustParse("100"),
+					Status: v1beta1.NodeClaimStatus{
+						ProviderID: test.RandomProviderID(),
+						Allocatable: map[v1.ResourceName]resource.Quantity{
+							v1.ResourceCPU:  resource.MustParse("5"),
+							v1.ResourcePods: resource.MustParse("100"),
+						},
 					},
-				},
-			})
-			// Block this pod from being disrupted with karpenter.sh/do-not-evict
-			pods[2].Annotations = lo.Assign(pods[2].Annotations, map[string]string{v1alpha5.DoNotEvictPodAnnotationKey: "true"})
+				})
 
-			ExpectApplied(ctx, env.Client, rs, pods[0], pods[1], pods[2], nodePool)
-			ExpectApplied(ctx, env.Client, nodeClaim, node, nodeClaim2, node2)
+				if spotToSpot {
+					nodeClaim2, node2 = test.NodeClaimAndNode(v1beta1.NodeClaim{
+						ObjectMeta: metav1.ObjectMeta{
+							Labels: map[string]string{
+								v1beta1.NodePoolLabelKey:     nodePool.Name,
+								v1.LabelInstanceTypeStable:   mostExpensiveSpotInstance.Name,
+								v1beta1.CapacityTypeLabelKey: mostExpensiveSpotOffering.CapacityType,
+								v1.LabelTopologyZone:         mostExpensiveSpotOffering.Zone,
+							},
+						},
+						Status: v1beta1.NodeClaimStatus{
+							ProviderID: test.RandomProviderID(),
+							Allocatable: map[v1.ResourceName]resource.Quantity{
+								v1.ResourceCPU:  resource.MustParse("5"),
+								v1.ResourcePods: resource.MustParse("100"),
+							},
+						},
+					})
+				}
```

Comment:
let me know if you think this is the same, but what I meant with the lo.Assign was to do this, since we only need to change the labels. 

```suggestion
				if spotToSpot {
				        nodeClaim2.Labels = lo.Assign(nodeClaim2.Labels map[string]string{
				                                 v1.LabelInstanceTypeStable:   mostExpensiveSpotInstance.Name,
								v1beta1.CapacityTypeLabelKey: mostExpensiveSpotOffering.CapacityType,
								v1.LabelTopologyZone:         mostExpensiveSpotOffering.Zone,
				        }
					 node2.Labels = lo.Assign(node2.Labels map[string]string{
				                                 v1.LabelInstanceTypeStable:   mostExpensiveSpotInstance.Name,
								v1beta1.CapacityTypeLabelKey: mostExpensiveSpotOffering.CapacityType,
								v1.LabelTopologyZone:         mostExpensiveSpotOffering.Zone,
				        }
				}
```

Reply:
Done.